rocAL/source/decoders/video/rocdec_video_decoder.cpp

```diff
@@ -0,0 +1,182 @@
+            if (flush_mode == ReconfigFlushMode::RECONFIG_FLUSH_MODE_DUMP_TO_FILE) {
+                ReconfigDumpFileStruct *p_dump_file_struct = static_cast<ReconfigDumpFileStruct *>(p_user_struct);
+                if (p_dump_file_struct->b_dump_frames_to_file) {
+                    viddec->SaveFrameToFile(p_dump_file_struct->output_file_name, pframe, surf_info);
+                }
+            }
+        }
+        // release and flush frame
+        viddec->ReleaseFrame(pts, true);
+        n_frames_flushed ++;
+    }
+
+    return n_frames_flushed;
+}
+
+// Initialize will open a new decoder and initialize the context
+VideoDecoder::Status RocDecVideoDecoder::Initialize(const char *src_filename, int device_id) {
+
+    VideoDecoder::Status status = Status::OK;
+    _src_filename = src_filename;
+    _device_id = device_id;
+    
+    // create rocDecoder and Demuxer for rocDecode
+    OutputSurfaceMemoryType mem_type = OUT_SURFACE_MEM_DEV_INTERNAL;      // set to internal
+    _demuxer = std::make_shared<VideoDemuxer>(src_filename);
+    rocDecVideoCodec rocdec_codec_id = AVCodec2RocDecVideoCodec(_demuxer->GetCodecID());
+    _rocvid_decoder = std::make_shared<RocVideoDecoder>(device_id, mem_type, rocdec_codec_id, 0, nullptr, 0);
+
+    if(!_rocvid_decoder->CodecSupported(device_id, rocdec_codec_id, _demuxer->GetBitDepth())) {
+        WRN("GPU doesn't support codec!")
+        return VideoDecoder::Status::FAILED;
+    }
+
+    std::string device_name, gcn_arch_name;
+    int pci_bus_id, pci_domain_id, pci_device_id;
+    _rocvid_decoder->GetDeviceinfo(device_name, gcn_arch_name, pci_bus_id, pci_domain_id, pci_device_id);
+    std::cout << "info: Using GPU device " << device_id << " - " << device_name << "[" << gcn_arch_name << "] on PCI bus " <<
+    std::setfill('0') << std::setw(2) << std::right << std::hex << pci_bus_id << ":" << std::setfill('0') << std::setw(2) <<
+    std::right << std::hex << pci_domain_id << "." << pci_device_id << std::dec << std::endl;
+
+    return status;
+}
+
+// Seeks to the frame_number in the video file and decodes each frame in the sequence.
+VideoDecoder::Status RocDecVideoDecoder::Decode(unsigned char *out_buffer, unsigned seek_frame_number, size_t sequence_length, size_t stride, int out_width, int out_height, int out_stride, AVPixelFormat out_pix_format) {
+    
+    VideoDecoder::Status status = Status::OK;
+    VideoSeekContext video_seek_ctx;
+
+    // Reconfig the decoder
+    ReconfigDumpFileStruct reconfig_user_struct = { 0 };
+    ReconfigParams reconfig_params = { 0 };
+    reconfig_params.p_fn_reconfigure_flush = ReconfigureFlushCallback;
+    reconfig_params.reconfig_flush_mode = RECONFIG_FLUSH_MODE_NONE;
+    reconfig_params.p_reconfig_user_struct = &reconfig_user_struct;
+    _rocvid_decoder->SetReconfigParams(&reconfig_params);
+
+    if (!_demuxer || !_rocvid_decoder || !out_buffer) {
+        ERR("RocDecVideoDecoder::Decoder is not initialized");
```

Comment:
Please check if we are handling the return statements. Will ERR break after printing the message?
If so return is not relevant.

Reply:
ERR is just internally std::cerr statement, hence if it fails we are printing the error and returning fail status

Reply:
From where are you calling this function? Is the return statement handled properly there?

Reply:
This is called from video_Read_and_Decode.cpp
https://github.com/fiona-gladwin/rocAL/blob/9239eb47a25faa56a9dcf480016b3158c9fd09a7/rocAL/source/loaders/video/video_read_and_decode.cpp#L122